R/run_grid.R

```diff
@@ -69,6 +69,191 @@ run_grid  <- function(gono_params, init_params, cost_params,
+##' `pv_red_net_cost` = present value of the reduction in net costs compared to
+##' baseline (i.e. sum of annual reduction in costs discounted at rate
+##'  `disc_rate` to time 0)
+##' `pv_qaly_gain` = present value of the QALYs gained due to vaccination
+##' compared to baseline (i.e. QALYs gained each year discounted at rate
+##'  `disc_rate` to time 0, and summed)
+##' `cet_20k` = cost effectiveness threshold (i.e. £ value per dose at which
+##' vaccination becomes cost-effective) calculated using £20,000 / QALY.
+##' (pv_red_net_cost + pv_qaly_gain * £20,000) / pv_inc_doses
+##' `cet_30k` = cost effectiveness threshold (i.e. £ value per dose at which
+##' vaccination becomes cost-effective) calculated using £30,000 / QALY
+##' (pv_red_net_cost + pv_qaly_gain * £30,000) / pv_inc_doses
+##' `inc_costs_9` = present value of incremental costs assuming £9 / dose.
+##' Incremental costs are calculated as: pv_inc_doses * £9 - pv_red_net_cost
+##' `inc_costs_18` = present value of incremental costs assuming £18 / dose.
+##' Incremental costs are calculated as: pv_inc_doses * £18 - pv_red_net_cost
+##' `inc_costs_50` = present value of incremental costs assuming £50 / dose.
+##' Incremental costs are calculated as: pv_inc_doses * £50 - pv_red_net_cost
+##' `inc_costs_85` = present value of incremental costs assuming £85 / dose
+##' Incremental costs are calculated as: pv_inc_doses * £85 - pv_red_net_cost
+##' @export
+compare_baseline_xpvwrh <- function(y, baseline, uptake_first_dose,
+                             uptake_second_dose, cost_params,
+                             disc_rate) {
+  ## compare run to baseline
+  flows <- extract_flows_xpvwrh(y)
+  ret <- Map(`-`, flows, baseline[names(flows)])
+  names(ret) <- paste0("inc_", names(flows))
+  ret <- c(flows, ret)
+
+  ## calculate number receiving primary vaccination
+  ret$inc_primary <- ret$inc_primary_total - ret$inc_vbe
+  ret$inc_cum_primary <- apply(ret$inc_primary, 2, cumsum)
+
+  ## calculate number receiving partial to full vaccination
+  ret$inc_cum_part_to_full <- apply(ret$inc_part_to_full,
+                                    2, cumsum)
+
+  ## cumulative individuals receiving booster vaccination (revaccination)
+  ret$inc_cum_revaccinated <- apply(ret$inc_revaccinated, 2, cumsum)
+
+  ## calculate cases averted per dose, both with and without discounting
+  ## return incremental annual and cumulative doses
+
+  # all vbe get two doses
+  ret$inc_vbe_doses <- ret$inc_vbe * 2
+
+  # calculate doses given per person offered primary vaccination
+  ret$inc_primary_full_doses <- ret$inc_primary * uptake_first_dose *
+                                                        uptake_second_dose
+  ret$inc_primary_part_doses <- ret$inc_primary * uptake_first_dose *
+                                                        (1 - uptake_second_dose)
+  ret$inc_primary_total_doses <- ret$inc_primary_full_doses +
+                                            ret$inc_primary_part_doses
```

Comment:
```suggestion
  ret$inc_primary_full_doses <- ret$inc_primary *  uptake_second_dose * 2
  ret$inc_primary_part_doses <- ret$inc_primary * (1 - uptake_second_dose)
  ret$inc_primary_total_doses <- ret$inc_primary_full_doses + ret$inc_primary_part_doses
```
first dose uptake should already be applied here, i.e. the number of peope moving from X to either P or V is number offered * first dose uptake. Then it's second dose uptake that determines whether they get 1 dose or 2.

Also, I don't think you multiplied the full doses by 2